### PR TITLE
Added secondary application compiles, untested

### PR DESCRIPTION
**Pull request by psmithcrl**
Some controls to help evaluate and test the functionality of the secondary application.
The primary goal being this will serve as an example for secondary application developers to use as a template for their application.

### CODE REVIEW

File: source/LibMultiSense/include/MultiSense/MultiSenseTypes.hh [hunk 1]
```diff
@@ -2976,6 +2977,57 @@ typedef void (*Callback)(const Header& header,
...
+public:
+
+    /** DataSource corresponding to secondaryAppDataP*/
+    DataSource  source;
+    /** Bits per pixel in the secondaryAppData */
+    uint32_t    bitsPerPixel;
+    /** Width of the secondaryAppData */
+    uint32_t    width;
+    /** Height of the secondaryAppData*/
+    uint32_t    height;
```
Comment (mattalvarado) [inline source/LibMultiSense/include/MultiSense/MultiSenseTypes.hh:2997]: Do these fields make sense? Is a void* with a length, associated frameID, and timestamp sufficient?

Comment (psmithcrl) [reply source/LibMultiSense/include/MultiSense/MultiSenseTypes.hh:2997]: No, I don't think hardly anything here makes sense yet. 
I am kind of using this as a template and a test utility to evaluate running  a secondary application on the camera.
I am going to change these fields, to make a basic demo.

Comment (mattalvarado) [reply source/LibMultiSense/include/MultiSense/MultiSenseTypes.hh:2997]: I think if we are going to support sending arbitrary number of bytes, we can let the user manually unpack it.

File: source/LibMultiSense/include/MultiSense/MultiSenseTypes.hh [hunk 2]
```diff
@@ -2976,6 +2977,49 @@ typedef void (*Callback)(const Header& header,
...
+    /** Unique ID used to describe an secondaryAppData. FrameIds increase sequentally from the device */
+    int64_t     frameId;
+    /** The time seconds value corresponding to when  the secondaryAppData was captured*/
+    uint32_t    timeSeconds;
+    /** The time microseconds value corresponding to when the secondaryAppData was captured*/
+    uint32_t    timeMicroSeconds;
+    /** The number of frames per second currently streaming from the device */
+    float       framesPerSecond;
+    /** The length of the secondaryAppData data stored in secondaryAppDataDataP */
+    uint32_t    secondaryAppDataLength;
```
Comment (mattalvarado) [inline source/LibMultiSense/include/MultiSense/MultiSenseTypes.hh:3005]: Is this redundant with the `length` member?

**Review by mattalvarado — APPROVED** [review]

**Pull request merged**